### PR TITLE
Made the show_matching_char highlighter use the matching_pairs option

### PR DESCRIPTION
**Pull request by laelath**
Implement the idea in #2436

### CODE REVIEW

File: src/highlighters.cc [hunk 1]
```diff
@@ -1130,52 +1130,56 @@ constexpr StringView LineNumbersHighlighter::ms_id;
 void show_matching_char(HighlightContext context, DisplayBuffer& display_buffer, BufferRange)
 {
     const Face face = context.context.faces()["MatchingChar"];
-    using CodepointPair = std::pair<Codepoint, Codepoint>;
-    static const CodepointPair matching_chars[] = { { '(', ')' }, { '{', '}' }, { '[', ']' }, { '<', '>' } };
+    // static const CodepointPair matching_chars[] = { { '(', ')' }, { '{', '}' }, { '[', ']' }, { '<', '>' } };
```
Comment (Screwtapello) [inline src/highlighters.cc]: You should remove this line, rather than just commenting it out.

File: src/highlighters.cc [hunk 2]
```diff
@@ -1130,52 +1130,56 @@ constexpr StringView LineNumbersHighlighter::ms_id;
...
     const auto& buffer = context.context.buffer();
     for (auto& sel : context.context.selections())
     {
         auto pos = sel.cursor();
         if (pos < range.begin or pos >= range.end)
             continue;
-        auto c = buffer.byte_at(pos);
-        for (auto& pair : matching_chars)
+        auto match = matching_pairs.end();
+        match = find(matching_pairs, buffer.byte_at(sel.cursor()));
```
Comment (mawww) [inline src/highlighters.cc]: Look for the codepoint, not the byte, so that arbitrary matching pairs can be used

File: src/highlighters.cc (hunk 1 above)
Comment (laelath) [reply src/highlighters.cc]: Oh that's embarrassing.

File: src/highlighters.cc (hunk 2 above)
Comment (laelath) [reply src/highlighters.cc]: Am I right that dereferencing a buffer iterator at the cursor position would do that? I admit that I have not looked too closely at the internal buffer manipulation.

File: src/highlighters.cc [hunk 3]
```diff
@@ -1130,52 +1130,55 @@ constexpr StringView LineNumbersHighlighter::ms_id;
...
     const auto& buffer = context.context.buffer();
     for (auto& sel : context.context.selections())
     {
         auto pos = sel.cursor();
         if (pos < range.begin or pos >= range.end)
             continue;
-        auto c = buffer.byte_at(pos);
-        for (auto& pair : matching_chars)
+        auto match = matching_pairs.end();
+        match = find(matching_pairs, *get_iterator(buffer, pos));
```
Comment (lenormf) [inline src/highlighters.cc]: Why not just `auto match = find(…)` ?

File: src/highlighters.cc [hunk 4]
```diff
@@ -1130,52 +1130,54 @@ constexpr StringView LineNumbersHighlighter::ms_id;
...
     const auto range = display_buffer.range();
     const auto& buffer = context.context.buffer();
     for (auto& sel : context.context.selections())
     {
         auto pos = sel.cursor();
         if (pos < range.begin or pos >= range.end)
             continue;
-        auto c = buffer.byte_at(pos);
-        for (auto& pair : matching_chars)
+        auto match = find(matching_pairs, *get_iterator(buffer, pos));
```
Comment (mawww) [inline src/highlighters.cc]: utf8::codepoint(get_iterator(buffer, pos), buffer.end());

**Pull request merged**